Makefile

```diff
@@ -91,13 +102,6 @@ vet:
 generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
```

Comment:
[L134-47](https://github.com/openshift/route-monitor-operator/pull/82/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R134-R147) (which I can't comment on directly): You should be able to remove the `mockgen` target, as we [provide it](https://github.com/openshift/boilerplate/blob/459c3cdf9d1bf451f99a08fbc21833c817d540e1/config/build.sh#L59-L63) in the boilerplate backing image.

(We would be happy to update that image to include [`controller-gen`](https://github.com/openshift/route-monitor-operator/pull/82/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R119) and [`kustomize`](https://github.com/openshift/route-monitor-operator/pull/82/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R149) as well.)

Reply:
the image doesn't seem to be used in the pipeline currently. Would be nice if `controller-gen`, `kustomize` and `operator-sdk` v1 were included there as well, probably a env variable could decide which version of the SDK to use

Reply:
Guess we'll switch to that image as soon as it's supported, but it's nothing that we should worry about now.

Reply:
> the image doesn't seem to be used in the pipeline currently.

Ah, right, that'll come in when we standardize prow config. See for example [this recent PR](https://github.com/openshift/release/pull/15470) for managed-upgrade-operator. It's [this bit](https://github.com/openshift/release/pull/15470/files#diff-06907a927697b3256104433ace45af68ffecb69ba5e61f421c692166ab151758R2) that makes prow pay attention to the [.ci-operator config](https://github.com/openshift/route-monitor-operator/pull/82/files#diff-4d5699538541ad6da5f6ae79150b49a432b610f4cba488cec683ac9dd00b18d2) in this PR, which is what grabs that image.

> Would be nice if `controller-gen`, `kustomize` and `operator-sdk` v1 were included there as well

Agreed. That would all go in [this script](https://github.com/openshift/boilerplate/blob/master/config/build.sh), which has several lovely examples to follow.

Specifically for osdk, we install all the supported versions in the image (via that script). The make targets run them via [this script](https://github.com/openshift/boilerplate/blob/459c3cdf9d1bf451f99a08fbc21833c817d540e1/boilerplate/openshift/golang-osd-operator/operator-sdk-generate.sh), which would need to be modified to include a condition for v1 with the appropriate syntax. Also see below:

> probably a env variable could decide which version of the SDK to use

The way it works today, we [discover your desired osdk version via go.mod](https://github.com/openshift/boilerplate/blob/459c3cdf9d1bf451f99a08fbc21833c817d540e1/boilerplate/openshift/golang-osd-operator/ensure.sh#L41). IIUC with v1 there's not actually a go dependency on osdk itself, so we'll need to either a) add an `indirect` one; or b) figure out some other mechanism.